config/routes.rb

```diff
@@ -1,13 +1,19 @@
+# frozen_string_literal: true
+
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
   # Defines the root path route ("/")
   # root "articles#index"
   root to: 'users#index'
 
-  get "/register", to: "users#new", as: :register
+  get '/register', to: 'users#new', as: :register
   resources :users, only: [:index, :new, :show, :create] do
-    resources :viewing_parties, only: [:show]
+    resources :discover, only: [:index]
+    resources :movies, only: [:index] do
+      resources :viewing_parties, only: [:show]
+    end
```

Comment:
This refactor using advanced routing looks really good and the resources make a lot of sense.